components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/HttpClientManager.java

```diff
@@ -0,0 +1,227 @@
+            // Ignore. If there was error while getting the property, continue with default value.
+        }
+
+        int connectionTimeout = defaultTimeout;
+        int readTimeout = defaultTimeout;
+        int connectionRequestTimeout = defaultTimeout;
+
+        if (connectionTimeoutString != null) {
+            try {
+                connectionTimeout = Integer.parseInt(connectionTimeoutString);
+            } catch (NumberFormatException e) {
+                LOG.error("Error while parsing connection timeout : " + connectionTimeoutString, e);
+            }
+        }
+        if (readTimeoutString != null) {
+            try {
+                readTimeout = Integer.parseInt(readTimeoutString);
+            } catch (NumberFormatException e) {
+                LOG.error("Error while parsing read timeout : " + connectionTimeoutString, e);
+            }
+        }
+        if (connectionRequestTimeoutString != null) {
+            try {
+                connectionRequestTimeout = Integer.parseInt(connectionRequestTimeoutString);
+            } catch (NumberFormatException e) {
+                LOG.error("Error while parsing connection request timeout : " + connectionTimeoutString, e);
+            }
+        }
+
+        return RequestConfig.custom()
+                .setConnectTimeout(connectionTimeout)
+                .setConnectionRequestTimeout(connectionRequestTimeout)
+                .setSocketTimeout(readTimeout)
+                .setRedirectsEnabled(false)
+                .setRelativeRedirectsAllowed(false)
+                .build();
+    }
+
+    private PoolingNHttpClientConnectionManager createPoolingConnectionManager() throws FrameworkException {
+
+        String maxConnectionsString = IdentityUtil.getProperty(Constants.CONNECTION_POOL_MAX_CONNECTIONS);
+        String maxConnectionsPerRouteString = IdentityUtil.getProperty(Constants
+                .CONNECTION_POOL_MAX_CONNECTIONS_PER_ROUTE);
+        int defaultMaxConnections = 20;
+        int maxConnections = defaultMaxConnections;
+        int maxConnectionsPerRoute = defaultMaxConnections;
+        try {
+            maxConnections = Integer.parseInt(maxConnectionsString);
+        } catch (NumberFormatException e) {
+            // Ignore. Default value is used.
+        }
+        try {
+            maxConnectionsPerRoute = Integer.parseInt(maxConnectionsPerRouteString);
+        } catch (NumberFormatException e) {
+            // Ignore. Default value is used.
+        }
+
+        ConnectingIOReactor ioReactor;
+        try {
+            ioReactor = new DefaultConnectingIOReactor();
+        } catch (IOReactorException e) {
+            throw new FrameworkException("Error while creating ConnectingIOReactor", e);
+        }
+        PoolingNHttpClientConnectionManager poolingHttpClientConnectionManager = new
+                PoolingNHttpClientConnectionManager(ioReactor);
+        // Increase max total connection to 50
```

Comment:
Add a fullstop.

Reply:
Resolved in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/110